control-plane-operator/controllers/hostedcontrolplane/cloud/kubevirt/reconcile.go

```diff
@@ -0,0 +1,201 @@
+	if err != nil {
+		return fmt.Errorf("failed to serialize cloudconfig: %w", err)
+	}
+
+	if cm.Data == nil {
+		cm.Data = map[string]string{}
+	}
+	cm.Data[CloudConfigKey] = string(serializedCfg)
+
+	return nil
+}
+
+func ReconcileCCMServiceAccount(sa *corev1.ServiceAccount, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(sa)
+	return nil
+}
+
+func ReconcileCCMRole(role *rbacv1.Role, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(role)
+	role.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{"kubevirt.io"},
+			Resources: []string{"virtualmachines"},
+			Verbs: []string{
+				"get",
+				"list",
+				"watch",
+			},
+		},
+		{
+			APIGroups: []string{"kubevirt.io"},
+			Resources: []string{"virtualmachineinstances"},
+			Verbs: []string{
+				"get",
+				"list",
+				"watch",
+				"update",
+			},
+		},
+		{
+			APIGroups: []string{""},
+			Resources: []string{"services"},
+			Verbs:     []string{rbacv1.VerbAll},
+		},
+	}
+	return nil
+}
+
+func ReconcileCCMRoleBinding(roleBinding *rbacv1.RoleBinding, ownerRef config.OwnerRef, sa *corev1.ServiceAccount, role *rbacv1.Role) error {
+	ownerRef.ApplyTo(roleBinding)
+	roleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "Role",
+		Name:     role.Name,
+	}
+	roleBinding.Subjects = []rbacv1.Subject{
+		{
+			Namespace: sa.Namespace,
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      sa.Name,
+		},
+	}
+	return nil
+}
+
+func ReconcileDeployment(deployment *appsv1.Deployment, hcp *hyperv1.HostedControlPlane, serviceAccountName string, releaseImage *releaseinfo.ReleaseImage) error {
+	clusterName, ok := hcp.Labels["cluster.x-k8s.io/cluster-name"]
+	if !ok {
+		return fmt.Errorf("\"cluster.x-k8s.io/cluster-name\" label doesn't exist in HostedControlPlane")
+	}
+	deploymentConfig := newDeploymentConfig()
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: ccmLabels(),
+		},
+		Strategy: appsv1.DeploymentStrategy{
+			Type: appsv1.RecreateDeploymentStrategyType,
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: ccmLabels(),
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					util.BuildContainer(CCMContainer(), buildCCMContainer(clusterName, releaseImage)),
+				},
+				Volumes:            []corev1.Volume{},
+				ServiceAccountName: serviceAccountName,
+			},
+		},
+	}
+
+	addVolumes(deployment)
+
+	config.OwnerRefFrom(hcp).ApplyTo(deployment)
+	deploymentConfig.ApplyTo(deployment)
+	return nil
+}
+
+func addVolumes(deployment *appsv1.Deployment) {
+
+	deployment.Spec.Template.Spec.Volumes = append(
+		deployment.Spec.Template.Spec.Volumes,
+		util.BuildVolume(ccmVolumeKubeconfig(), buildCCMVolumeKubeconfig),
+	)
+	deployment.Spec.Template.Spec.Volumes = append(
+		deployment.Spec.Template.Spec.Volumes,
+		util.BuildVolume(ccmCloudConfig(), buildCCMCloudConfig),
+	)
+}
+
+func getPodVolumeMounts() util.PodVolumeMounts {
```

Comment:
More idiomatic golang would be like

```suggestion
func podVolumeMounts() util.PodVolumeMounts {
```

Reply:
Changed, thanks